src/parsers/perf/perfparser.cpp

```diff
@@ -863,24 +863,38 @@ struct PerfParserPrivate
         setParents(&topDownResult.children, nullptr);
     }
 
+    static void buildCallerCalleeResult(const QVector<FrameData>& bottomUpData, QVector<FrameData>& callerCalleeData)
+    {
+        static QSet<QString> callerCalleeRecursionGuard;
+
+        for (const FrameData& row : bottomUpData) {
+            if (!callerCalleeRecursionGuard.contains(row.location)) {
+                FrameData newFrame;
+                newFrame.symbol = row.symbol;
+                newFrame.binary = row.binary;
+                newFrame.location = row.location;
+                newFrame.address = row.address;
+                newFrame.selfCost = 1;
+                newFrame.inclusiveCost = 1;
+                callerCalleeData.push_back(newFrame);
+                callerCalleeRecursionGuard.insert(row.location);
+            } else {
+                for (FrameData& callerCalleeRow : callerCalleeData) {
+                    if (callerCalleeRow.location == row.location) {
+                        callerCalleeRow.inclusiveCost++;
```

Comment:
this is still potentially counting the same thing multiple times, i.e. when you have a recursive stack

    a > b > a > b

then the first two frames will not yet be in the set and thus added to the list, but then the last two frames will get their inclusive cost incremented while that shouldn't be the case for the caller/callee data representation.